client/lib/browser-detection/index.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * External dependencies
+ */
+import { isUndefined } from 'lodash';
+
+export const isFirefox = ! isUndefined( window.InstallTrigger );
+
+export const isIE11 = (
+	window &&
+	window.MSInputMethodContext &&
+	document &&
+	document.documentMode
+);
```

Comment:
I like this module but I feel like we _really_ need some links to the documentation stating that this kind of detection will work. how reliable is it? _when_ was this a reliable detection method, etc…